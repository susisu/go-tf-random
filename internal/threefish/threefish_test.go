package threefish_test

import (
	"os"
	"testing"

	"github.com/gkampitakis/go-snaps/snaps"
	tf "github.com/susisu/go-tf-random/internal/threefish"
)

func TestMain(t *testing.M) {
	v := t.Run()
	snaps.Clean(t)
	os.Exit(v)
}

func TestThreefish256EncryptBlock64(t *testing.T) {
	key := []uint64{
		0x00000000_00000000,
		0x89abcdef_01234567,
		0x01234567_89abcdef,
		0xffffffff_ffffffff,
	}
	block := []uint64{
		0x00000000_00000000,
		0x89abcdef_01234567,
		0x01234567_89abcdef,
		0xffffffff_ffffffff,
	}
	out := tf.Threefish256EncryptBlock64(key, block)
	snaps.MatchSnapshot(t, out)
}

func TestThreefish256EncryptBlock32(t *testing.T) {
	key := []uint64{
		0x00000000_00000000,
		0x89abcdef_01234567,
		0x01234567_89abcdef,
		0xffffffff_ffffffff,
	}
	block := []uint64{
		0x00000000_00000000,
		0x89abcdef_01234567,
		0x01234567_89abcdef,
		0xffffffff_ffffffff,
	}
	out := tf.Threefish256EncryptBlock32(key, block)
	snaps.MatchSnapshot(t, out)
}
