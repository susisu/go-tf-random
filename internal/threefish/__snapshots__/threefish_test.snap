
[TestThreefish256EncryptBlock64 - 1]
threefish.Uint64x4{0x1fa513db1451b1a, 0x587b6f21e58ebd5b, 0xa5ed40f05b2bb82f, 0x4b8671bf18e5b514}
---

[TestThreefish256EncryptBlock32 - 1]
threefish.Uint32x8{0x1fa513d, 0xb1451b1a, 0x587b6f21, 0xe58ebd5b, 0xa5ed40f0, 0x5b2bb82f, 0x4b8671bf, 0x18e5b514}
---
